stdlib/http/src/main/java/org/ballerinalang/net/http/websocket/WebSocketUtil.java

```diff
@@ -217,11 +256,447 @@ public static ErrorValue createErrorCause(String message, String reason, BPackag
+        }
+
+        long idleTimeoutInSeconds = WebSocketUtil.findIdleTimeoutInSeconds(clientEndpointConfig);
+        if (idleTimeoutInSeconds > 0) {
+            clientConnectorConfig.setIdleTimeoutInMillis((int) (idleTimeoutInSeconds * 1000));
+        }
+
+        clientConnectorConfig.setMaxFrameSize(WebSocketUtil.findMaxFrameSize(clientEndpointConfig));
+
+        MapValue secureSocket = clientEndpointConfig.getMapValue(HttpConstants.ENDPOINT_CONFIG_SECURE_SOCKET);
+        if (secureSocket != null) {
+            HttpUtil.populateSSLConfiguration(clientConnectorConfig, secureSocket);
+        } else if (scheme.equals(WSS_SCHEME)) {
+            clientConnectorConfig.useJavaDefaults();
+        }
+        clientConnectorConfig.setWebSocketCompressionEnabled(
+                clientEndpointConfig.getBooleanValue(WebSocketConstants.COMPRESSION_ENABLED_CONFIG));
+    }
+
+    private static Map<String, String> getCustomHeaders(MapValue<String, Object> headers) {
+        Map<String, String> customHeaders = new HashMap<>();
+        headers.keySet().forEach(
+                key -> customHeaders.put(key, headers.get(key).toString())
+        );
+        return customHeaders;
+    }
+
+    /**
+     * Populate the retry config.
+     * @param retryConfig a retry config
+     * @param retryConnectorConfig a doReconnect connector config
+     */
+    public static void populateRetryConnectorConfig(MapValue<String, Object> retryConfig,
+                                                    RetryContext retryConnectorConfig) {
+        int interval = Integer.parseInt(retryConfig.get(INTERVAL).toString());
+        float backOfFactor = Float.parseFloat(retryConfig.get(BACK_OF_FACTOR).toString());
+        int maxAttempts = Integer.parseInt(retryConfig.get(MAX_COUNT).toString());
+        int maxInterval = Integer.parseInt(retryConfig.get(MAX_INTERVAL).toString());
+        if (interval < 0) {
+            logger.warn("The interval's value set for the configuration needs to be " +
+                    "greater than -1. The interval[" + interval + "] value is set to 1000");
+            interval = 1000;
+        }
+        if (backOfFactor < 0) {
+            logger.warn("The decay's value set for the configuration needs to be " +
+                    "greater than -1. The backOfFactor[" + backOfFactor + "] value is set to 1.0");
+            backOfFactor = (float) 1.0;
+        }
+        if (maxInterval < 0) {
+            logger.warn("The maxInterval's value set for the configuration needs to be " +
+                    "greater than -1. The maxInterval[" + maxInterval + "] value is set to 30000");
+            maxInterval =  30000;
+        }
+        if (maxAttempts < 0) {
+            logger.warn("The maximum doReconnect attempt's value set for the configuration needs to be " +
+                    "greater than -1. The maxAttempts[ " + maxAttempts + "] value is set to 0");
+            maxAttempts = 0;
+        }
```

Comment:
Seems like we are duplicating code here. Can't we use a single function to which we pass the params? We can have parsing the integer part also in that method. Try using the Constants when logging also.